src/fdopendir.c

```diff
@@ -43,8 +55,12 @@ DIR *fdopendir(int dirfd) {
         return 0;
     }
 
-    if (dirfd == AT_FDCWD)
-        return opendir (".");
+    if (dirfd == AT_FDCWD) {
+        dir = opendir (".");
+        /* dirfd can be closed only upon success */
+        if (dir && dirfd != -1) PROTECT_ERRNO(close(dirfd));
```

Comment:
We knows here that `dirfd != -1` (since it equals `AT_FDCWD`, which is -2 according to `sys/fcntl.h`), so that part can be removed.

Reply:
done